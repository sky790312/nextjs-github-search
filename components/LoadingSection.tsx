import React from 'react'
import styled from 'styled-components'
import { TextCenterContainer } from '@/GlobalStyles'
import { Spinner } from '@/components/Spinner'

export interface Props {
  shouldHide: boolean
  isLoading: boolean
  onClick: () => void
}

export const LoadingSection: React.FC<Props> = React.memo(
  ({ shouldHide, onClick, isLoading }) => {
    return (
      <LoadingSectionWrapper shouldHide={shouldHide}>
        {isLoading && <Spinner />}
        <Button disabled={isLoading} onClick={onClick}>
          Load More
        </Button>
      </LoadingSectionWrapper>
    )
  },
)

const LoadingSectionWrapper = styled(TextCenterContainer)<{
  shouldHide: boolean
}>`
  position: relative;
  padding: 10px;
  visibility: ${props => (props.shouldHide ? 'hidden' : 'visible')};
`

const Button = styled.button<{ disabled: boolean }>`
  opacity: ${props => (props.disabled ? '0.3' : '1')};
  cursor: ${props => (props.disabled ? 'default' : 'pointer')};
`
