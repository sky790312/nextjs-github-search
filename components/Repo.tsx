/* eslint-disable @typescript-eslint/camelcase */
import styled from 'styled-components'
import { GithubRepoSchema } from '@/api/schema'
import Link from 'next/link'
import { useRouter } from 'next/router'

export interface Props {
  repo: GithubRepoSchema
}

export const Repo: React.FC<Props> = ({ repo }) => {
  const router = useRouter()
  const { userName } = router.query

  return (
    <Link href={`/repo?userName=${userName}&repoName=${repo.name}`}>
      <RepoWrapper>
        <div>
          <h3>{repo.name}</h3>
          <p>Star: {repo.stargazers_count}</p>
        </div>
      </RepoWrapper>
    </Link>
  )
}

const RepoWrapper = styled.div`
  margin: 10px;
  padding: 20px;
  cursor: pointer;
  background-color: ${props => props.theme.colors.gray};

  &:hover {
    opacity: 0.7;
  }

  @media (min-width: 1140px) {
    display: flex;
    align-items: center;
  }
`
