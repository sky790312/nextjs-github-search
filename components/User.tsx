/* eslint-disable @typescript-eslint/camelcase */
import styled from 'styled-components'
import { GithubUserSchema } from '@/api/schema'
import Link from 'next/link'

export interface Props {
  user: GithubUserSchema
}

export const User: React.FC<Props> = ({ user }) => {
  return (
    <Link href={`/user?userName=${user.login}`}>
      <UserWrapper>
        <StyledUserAvatar src={user.avatar_url} />
        <div>
          <h3>{user.login}</h3>
        </div>
      </UserWrapper>
    </Link>
  )
}

const UserWrapper = styled.div`
  margin: 10px;
  padding: 20px;
  cursor: pointer;
  background-color: ${props => props.theme.colors.gray};

  &:hover {
    opacity: 0.7;
  }

  @media (min-width: 1140px) {
    display: flex;
    align-items: center;
  }
`

const StyledUserAvatar = styled.img`
  width: 100%;

  @media (min-width: 1140px) {
    width: 200px;
    margin-right: 20px;
  }
`
