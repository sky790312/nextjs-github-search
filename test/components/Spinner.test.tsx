import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import { Spinner } from '@/components/Spinner'

describe('Spinner component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(withProviders(<Spinner />), {})
    expect(asFragment()).toMatchSnapshot()
  })
})
