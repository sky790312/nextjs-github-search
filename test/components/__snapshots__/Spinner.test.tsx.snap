// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="Spinner__StyledSpinner-sc-12ylinb-0 berLTj"
  />
</DocumentFragment>
`;
